neuraxle/base.py

```diff
@@ -33,6 +33,7 @@
 
 from conv import convolved_1d
 from joblib import dump, load
+from sklearn.base import BaseEstimator
```

Comment:
I don't like importing sklearn in neuraxle.base. What will happen when we'll support other libraries? We should maybe never be allowed to truly import them to avoid bloating the requirements. Scikit-learn is perhaps the exception.